### ISSUE TITLE
Image fields not available in Studio when editing subpanels

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM#1305]
It seems the images for SugarField Image fields aren't available to be included in subpanels when we're editing these in Studio.

Steps to reproduce:
1. In Studio, Add an image field to some module (ex. Contacts).
2. In Studio, try to add that field to some subpanel that has contacts (e.g. a subpanel under Accounts).
3. Field isn't there to be included.

However, adding this field manually is as easy as going into the appropriate custom file, like this

`custom/modules/Contacts/metadata/subpanels/ForAccounts.php`

and adding the following

```
'photo' =>array (
                'name' => 'photo',
                'vname' => 'LBL_PHOTO',
                'width' => '6%',
                'default' => true,
        ),
```

So one would expect Studio to be able to handle it.

Image fields seem to be ready for list views and subpanels, since there is a file handling them in 

`include/SugarFields/Fields/ImageListView.tpl`

However, another bug appears here; I'm filing it separately.

**Comment by pgorod** [comment]
Every once in a while this comes up in the forums.

This could be as simple as turning on a flag to make this field appear in Studio... only it would have to be someone familiar with the Studio code to do the fix. I'm afraid it's a very complicated area of SuiteCRM code for someone who goes looking in there for the first time.

**Comment by ijdavie** [comment]
@pgorod, This is by design i am afraid, however you can add the field into the subpanel/list view by adding something like this to the vardef of the given image field.  

'studio' => array('listview' => true),

a repair and rebuild and it should allow users to add these to both subpanel and list views.

**Comment by pgorod** [comment]
@ijdavie thanks for the follow-up. If you don't mind my asking, why is this "by design"? Shouldn't we be reducing the number of times people have to do things in code when they could be using Studio instead?

Why not make that vardef change apply to every field of this type that gets added?

**Comment by ijdavie** [comment]
@pgorod, This I dont completely know why. seems to be in the core code that came from SugarCRM, where without this line above it will not work as it removes it from the field defs array on listview and subpanel views. Only thing I can think of is it is down to the image field possibly affecting the layout.

**Comment by pgorod** [comment]
@ijdavie Yes it does seem like they deactivated it. I wonder if we can't simply activate it. My experience is that this looks pretty good in the subpanels, and it really helps usability for some kinds of data. 

I'm using it for a list of people. We have photos for all of them. It works great, the layout looks fine and users love it. I'm sure there are tons of people out there who could use this for lists of products, etc.

If you agree maybe we could leave this open as an enhancement request. But it's your call, I'm ok with whatever you think is best for the project. Thanks.

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Image fields not available in Studio when editing subpanels](https://trello.com/c/T84q8pfG/101-image-fields-not-available-in-studio-when-editing-subpanels)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
